### PR TITLE
Allow renaming cluster, host, and storage

### PR DESCRIPTION
**Pull request by GabrielBrascher**
## Description

This PR adds implementation for changing host and storage name, additionally, it fixes a Bug on cluster updateCluster API command. This PRs also enhances the UI by allowing editing field name on Host and Storage pool. Due to the fact that there is no support to editing cluster via UI, it was not edited.

TODO:  I will address Host,  Cluster, and Storage Pool name edition on CloudStack Primate once the API implementation gets merged.

**Details:** 
Prior to this PR the following API commands did not offer support for updating name:
- [updateHost](https://cloudstack.apache.org/api/apidocs-4.14/apis/updateHost.html) **(enhancement)**
- [updateStoragePool](https://cloudstack.apache.org/api/apidocs-4.14/apis/updateStoragePool.html)  **(enhancement)**

Additionally, [updateCluster](https://cloudstack.apache.org/api/apidocs-4.14/apis/updateCluster.html) claims to support changing a cluster name (via `clustername` parameter); however, such operation did not work.  **(bug)**

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

- Via API,  change host, cluster, and storage pool names.
- Via UI, change host and storage pool names.

### CODE REVIEW

File: server/src/main/java/com/cloud/resource/RollingMaintenanceManagerImpl.java [hunk 1]
```diff
@@ -114,12 +115,15 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
         return true;
     }
 
-    private void updateCluster(long clusterId, String state) {
+    private void updateCluster(long clusterId, String allocationState) {
```
Comment (GabrielBrascher) [inline server/src/main/java/com/cloud/resource/RollingMaintenanceManagerImpl.java:118]: @nvazquez can you please take a look at it?

I changed API parameter handling from separate parameters on the method call to an object of UpdateClusterCmd class. Therefore, I needed to change this tiny piece of RollingMaintenanceManagerImpl.

**Comment by wido** [comment]
Looks good to me based on the code

**Comment by rhtyd** [comment]
Renaming of cluster on VMware could cause a problem if the cluster on VCenter is not mapped; on XenServer I'm not sure if there'll be any side-effects. On KVM LGTM

File: server/src/main/java/com/cloud/resource/RollingMaintenanceManagerImpl.java (hunk 1 above)
Comment (nvazquez) [reply server/src/main/java/com/cloud/resource/RollingMaintenanceManagerImpl.java:118]: Thanks @GabrielBrascher, looks good to me. The change needed in RollingMaintenanceManagerImpl will be just the same as you've done below

**Comment by GabrielBrascher** [comment]
@nvazquez thanks for the review!

I think that I already did the necessary changes in the method `updateCluster` at `RollingMaintenanceManagerImpl`. Please let me know if I am missing something.

**Comment by GabrielBrascher** [comment]
@rhtyd thanks for the feedback, I will prevent cluster renaming on VMware then (maybe on Xenserver as well).

**Comment by rhtyd** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java [hunk 2]
```diff
@@ -1026,12 +1027,26 @@ public void doInTransactionWithoutResult(final TransactionStatus status) {
...
+        String allocationState = cmd.getAllocationState();
+        String managedstate = cmd.getManagedstate();
+        String name = cmd.getClusterName();
 
-        final ClusterVO cluster = (ClusterVO)clusterToUpdate;
         // Verify cluster information and update the cluster if needed
         boolean doUpdate = false;
 
+        if (org.apache.commons.lang.StringUtils.isNotBlank(name)) {
+            if(cluster.getHypervisorType() == HypervisorType.VMware) {
```
Comment (GabrielBrascher) [inline server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:1042]: @rhtyd considering your conserns with issues caused by renaming VMware clusters I added this line.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Pull request merged**